blueprints/third-party-solutions/phpipam/glb.tf

```diff
@@ -0,0 +1,160 @@
+        domains = [local.domain]
+      }
+    }
+  }
+}
+
+# Cloud Armor configuration
+resource "google_compute_security_policy" "policy" {
+  count   = local.glb_create && var.security_policy.enabled ? 1 : 0
+  name    = "cloud-run-policy"
+  project = module.project.project_id
+  rule {
+    action   = "deny(403)"
+    priority = 1000
+    match {
+      versioned_expr = "SRC_IPS_V1"
+      config {
+        src_ip_ranges = var.security_policy.ip_blacklist
+      }
+    }
+    description = "Deny access to list of IPs"
+  }
+  rule {
+    action   = "deny(403)"
+    priority = 900
+    match {
+      expr {
+        expression = "request.path.matches(\"${var.security_policy.path_blocked}\")"
+      }
+    }
+    description = "Deny access to specific URL paths"
+  }
+  rule {
+    action   = "allow"
+    priority = "2147483647"
+    match {
+      versioned_expr = "SRC_IPS_V1"
+      config {
+        src_ip_ranges = ["*"]
+      }
+    }
+    description = "Default rule"
+  }
+}
+
+# Identity-Aware Proxy (IAP) or OAuth brand (see OAuth consent screen)
+# Note:
+# Only "Organization Internal" brands can be created programmatically
+# via API. To convert it into an external brand please use the GCP
+# Console.
+# Brands can only be created once for a Google Cloud project and the
+# underlying Google API doesn't support DELETE or PATCH methods.
+# Destroying a Terraform-managed Brand will remove it from state but
+# will not delete it from Google Cloud.
+resource "google_iap_brand" "iap_brand" {
+  count   = local.glb_create && var.iap.enabled ? 1 : 0
+  project = module.project.project_id
+  # Support email displayed on the OAuth consent screen. The caller must be
+  # the user with the associated email address, or if a group email is
+  # specified, the caller can be either a user or a service account which
+  # is an owner of the specified group in Cloud Identity.
+  support_email     = var.iap.email
+  application_title = var.iap.app_title
+}
+
+# IAP owned OAuth2 client
+# Note:
+# Only internal org clients can be created via declarative tools.
+# External clients must be manually created via the GCP console.
+# Warning:
+# All arguments including secret will be stored in the raw state as plain-text.
+resource "google_iap_client" "iap_client" {
+  count        = local.glb_create && var.iap.enabled ? 1 : 0
+  display_name = var.iap.oauth2_client_name
+  brand        = google_iap_brand.iap_brand[0].name
+}
+
+# IAM policy for IAP
+# For simplicity we use the same email as support_email and authorized member
+resource "google_iap_web_iam_member" "iap_iam" {
+  count   = local.glb_create && var.iap.enabled ? 1 : 0
+  project = module.project.project_id
+  role    = "roles/iap.httpsResourceAccessor"
+  member  = "user:${var.iap.email}"
+}
+
+# SA service agent for IAP, which invokes CR
+# Note:
+# Once created, this resource cannot be updated or destroyed. These actions are a no-op.
+resource "google_project_service_identity" "iap_sa" {
```

Comment:
the project module should do this for you automatically

Reply:
Removed resource and added iap.googleapis.com service

Reply:
Actually I think you need to add `jit:true` to the iap entry in [this file](https://github.com/GoogleCloudPlatform/cloud-foundation-fabric/blob/master/modules/project/service-agents.yaml)

Reply:
You were right, not adding jit would result in the service agent not being created, thanks a lot!